src/caret_analyze/architecture/architecture.py

```diff
@@ -466,6 +467,82 @@ def rename_topic(self, src: str, dst: str) -> None:
         for c in self._communications:
             c.rename_topic(src, dst)
 
+    @staticmethod
+    def diff_node_names(
+        left_arch: Architecture,
+        right_arch: Architecture
+    ) -> Tuple[Tuple[str, ...], Tuple[str, ...]]:
+        """
+        Compare two architecture objects (: left_arch, right_arch) and returns\
+        the node names that exist only in the respective architectures.
+
+        Parameters
+        ----------
+        left_arch : Architecture
+            Architecture object
+        right_arch : Architecture
+            Architecture object
+
```

Comment:
For all docstring:

The "Returns" field is missing.

Reference: 
https://github.com/tier4/CARET_analyze/blob/main/src/caret_analyze/architecture/combine_path.py#L46